packages/core/src/delegation/DelegationNode.ts

```diff
@@ -259,38 +258,38 @@ export class DelegationNode implements IDelegationNode {
    * This is required to anchor the delegation node on chain in order to enforce the delegate's consent.
    *
    * @param delegateDid The DID of the delegate.
-   * @param sign The callback to sign the delegation creation details for the delegate.
+   * @param signers An array of signer interfaces, one of which will be selected to sign the delegation creation details for the delegate.
    * @returns The DID signature over the delegation **as a hex string**.
    */
   public async delegateSign(
     delegateDid: DidDocument,
-    sign: SignCallback
+    signers: SignerInterface[]
   ): Promise<Did.EncodedSignature> {
-    const delegateSignature = await sign({
-      data: this.generateHash(),
-      did: delegateDid.id,
-      verificationRelationship: 'authentication',
-    })
-    const signerUrl =
-      `${delegateDid.id}${delegateSignature.verificationMethod.id}` as DidUrl
-    const { fragment } = Did.parse(signerUrl)
-    if (!fragment) {
-      throw new SDKErrors.DidError(
-        `DID verification method URL "${signerUrl}" couldn't be parsed`
-      )
-    }
-    const verificationMethod = delegateDid.verificationMethod?.find(
-      ({ id }) => id === fragment
+    const { byDid, verifiableOnChain } = Signers.select
```

Comment:
you `select` a `Signer` `byDid` (-> by it's association to a Did, via the DidDocument). I thought that was self-explanatory, how can this be improved?

Reply:
I think on its own it might not make sense, but when it's exported with a named object, it made more sense to me, also the way it's used in the tests.

Reply:
any suggestions here?